src/compiler/moduleNameResolver.ts

```diff
@@ -392,10 +392,11 @@ namespace ts {
         if (resolved) {
             const { fileName, packageId } = resolved;
             const resolvedFileName = options.preserveSymlinks ? fileName : realPath(fileName, host, traceEnabled);
+            const pathsAreEqual = arePathsEqual(fileName, resolvedFileName, host);
             resolvedTypeReferenceDirective = {
                 primary,
-                resolvedFileName,
-                originalPath: arePathsEqual(fileName, resolvedFileName, host) ? undefined : fileName,
+                resolvedFileName: pathsAreEqual ? fileName : resolvedFileName,

```

Comment:
Personally, I think this needs a comment so someone doesn't simplify it back to `resolvedFileName`.

Reply:
Perfect - thanks!